src/components/ApiKeyAuth/ApiKeyAuthFlow/ApiKeyAuthFlow.tsx

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from 'react';
+import { GenerateConnectionRequest } from '@generated/api/src';
+
+import { LandingContent } from 'components/ApiKeyAuth/ApiKeyAuthFlow/LandingContent';
+import { useApiKey } from 'context/ApiKeyContextProvider';
+import { useProject } from 'context/ProjectContextProvider';
+import { api, Connection } from 'services/api';
+
+type ApiKeyAuthFlowProps = {
+  provider: string;
+  consumerRef: string;
+  consumerName?: string;
+  groupRef: string;
+  groupName?: string;
+  children: JSX.Element,
+  setSelectedConnection: (connection: Connection) => void;
+};
+
+export function ApiKeyAuthFlow({
+  provider, consumerRef, consumerName, groupRef, groupName, children, setSelectedConnection,
+}: ApiKeyAuthFlowProps) {
+  const project = useProject();
+  const [connection, setConnection] = useState<Connection | null>(null);
+  const [nextStep, setNextStep] = useState<boolean>(false);
+  const [userApiKey, setUserApiKey] = useState<string | null>(null);
+  const apiKey = useApiKey();
+
+  useEffect(() => {
+    if (connection !== null) {
+      setSelectedConnection(connection);
```

Comment:
what's the reasoning behind setting the the connection in the local (setConnection) and parent state (setSelectedConnection)? Should we just keep track of state in the parent or is there case where we want to track them separately?